src/components/StaticFilters.tsx

```diff
@@ -1,43 +1,25 @@
-import { useMemo } from 'react';
 import { CompositionMethod } from '../hooks/useComposedCssClasses';
-import {
-  ApplyFiltersButton,
-  CheckboxOption,
-  CollapsibleLabel,
-  CollapsibleSection,
-  FilterGroup,
-  SearchInput
-} from './Filters';
+import { FilterGroup, FilterGroupCssClasses, FilterOptionConfig } from './FilterGroup';
 import { StaticFilters as StaticFiltersCompoundComponent } from './Filters/StaticFilters';
 
 /**
  * The CSS class interface for {@link StaticFilters}.
  *
  * @public
  */
-export interface StaticFiltersCssClasses {
-  container?: string,
-  searchInput?: string,
-  optionsContainer?: string,
-  option?: string,
-  optionInput?: string,
-  optionLabel?: string,
-  applyFiltersButton?: string
+export interface StaticFiltersCssClasses extends FilterGroupCssClasses {
+  container?: string
 }
 
 /**
- * The configuration data for a filter option.
+ * The configuration data for a static filter option.
  *
  * @public
  */
-export interface FilterOptionConfig {
-  /** {@inheritDoc Filters.CheckboxOptionProps.value} */
-  value: string | number | boolean,
-  /** {@inheritDoc Filters.CheckboxOptionProps.label} */
-  label?: string,
-  /** {@inheritDoc Filters.CheckboxOptionProps.selectedByDefault} */
-  selectedByDefault?: boolean
-}
+export type StaticFilterOptionConfig = Omit<FilterOptionConfig, 'matcher' | 'value'> & {
```

Comment:
I think sharing types across different values can be nice, but maybe we need to think about this one a little more? Currently `StaticFilterOptionConfig` is `Omit<FilterOptionConfig, 'matcher' | 'value'>` and then override `value` with a different type, and `FilterOptionConfig` is `Omit<CheckboxOptionProps, 'customCssClasses' | 'cssCompositionMethod'>`. I feel like that's a lot of mental gymnastics and isn't really a clear mental model

Reply:
Since the components build on top of each other, I think it's kind of nice to have the configs depends on each other. This way, it's easier to make changes and/or keep the props align for all these components. I can reorganize it a bit so it's more clear.